### PR TITLE
Small code cleanup and simplification

### PR DESCRIPTION
**Pull request by benoitsteiner**

### CODE REVIEW

File: apps/autoscheduler/FunctionDAG.cpp [hunk 1]
```diff
@@ -601,12 +601,11 @@ FunctionDAG::FunctionDAG(const vector<Function> &outputs, const MachineParams &p
...
         for (int j = 0; j < consumer.dimensions(); j++) {
-            Expr min_var = Variable::make(Int(32), consumer.name() + "." + consumer.args()[j] + ".min");
-            Expr max_var = Variable::make(Int(32), consumer.name() + "." + consumer.args()[j] + ".max");
-            Expr extent = max_var - min_var + 1;
+            Halide::Var min_var(consumer.name() + "." + consumer.args()[j] + ".min");
+            Halide::Var max_var(consumer.name() + "." + consumer.args()[j] + ".max");
             Interval interval(min_var, max_var);
             scope.push(consumer.args()[j], interval);
-            node.region_required.push_back(interval);
+            node.region_required.push_back(SymbolicRange{min_var, max_var});
```
Comment (steven-johnson) [inline apps/autoscheduler/FunctionDAG.cpp]: Would this be better as `emplace_back()`?

File: apps/autoscheduler/FunctionDAG.h [hunk 2]
```diff
@@ -348,6 +348,11 @@ struct FunctionDAG {
     // An edge is a producer-consumer relationship
     struct Edge;
 
+    struct SymbolicRange {
```
Comment (abadams) [inline apps/autoscheduler/FunctionDAG.h]: Halide::Range is min, extent, and Halide::Interval is min, max. In keeping with that convention, how about naming this SymbolicInterval?

Comment (benoitsteiner) [reply apps/autoscheduler/FunctionDAG.h]: Good idea, done

File: apps/autoscheduler/FunctionDAG.cpp (hunk 1 above)
Comment (benoitsteiner) [reply apps/autoscheduler/FunctionDAG.cpp]: Indeed, fixed

**Pull request merged**